ingestion-scheduler/src/main/java/com/datahub/metadata/ingestion/IngestionScheduler.java

```diff
@@ -212,6 +229,11 @@ public void run() {
 
             // 3. Reschedule ingestion sources based on the fetched schedules (inside "info")
             log.debug("Received batch of Ingestion Source Info aspects. Attempting to re-schedule execution requests.");
+
+            // First unschedule all currently scheduled runs (to make sure consistency is maintained)
+            _unscheduleAll.run();
```

Comment:
what if a run is in the middle of being executed? will it be unscheduled, or is this just runs that are set to run but have not started execution yet?

Reply:
Yep the second. If its in the middle, its still going to complete